packages/twenty-server/src/engine/core-modules/workspace-trusted-domain/workspace-trusted-domain.resolver.ts

```diff
@@ -0,0 +1,70 @@
+import { UseGuards } from '@nestjs/common';
+import { Args, Mutation, Resolver, Query } from '@nestjs/graphql';
+
+import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
+import { AuthWorkspace } from 'src/engine/decorators/auth/auth-workspace.decorator';
+import { WorkspaceAuthGuard } from 'src/engine/guards/workspace-auth.guard';
+import { WorkspaceTrustedDomainService } from 'src/engine/core-modules/workspace-trusted-domain/services/workspace-trusted-domain.service';
+import { WorkspaceTrustedDomain } from 'src/engine/core-modules/workspace-trusted-domain/dtos/trusted-domain.dto';
+import { CreateTrustedDomainInput } from 'src/engine/core-modules/workspace-trusted-domain/dtos/create-trusted-domain.input';
+import { AuthUser } from 'src/engine/decorators/auth/auth-user.decorator';
+import { User } from 'src/engine/core-modules/user/user.entity';
+import { DeleteTrustedDomainInput } from 'src/engine/core-modules/workspace-trusted-domain/dtos/delete-trusted-domain.input';
+import { ValidateTrustedDomainInput } from 'src/engine/core-modules/workspace-trusted-domain/dtos/validate-trusted-domain.input';
+
+@UseGuards(WorkspaceAuthGuard)
+@Resolver()
+export class WorkspaceTrustedDomainResolver {
+  constructor(
+    private readonly workspaceTrustedDomainService: WorkspaceTrustedDomainService,
+  ) {}
+
+  @Mutation(() => WorkspaceTrustedDomain)
+  async createWorkspaceTrustedDomain(
+    @Args('input') { domain, email }: CreateTrustedDomainInput,
+    @AuthWorkspace() currentWorkspace: Workspace,
+    @AuthUser() currentUser: User,
+  ): Promise<WorkspaceTrustedDomain> {
+    return this.workspaceTrustedDomainService.createTrustedDomain(
+      domain,
+      currentWorkspace,
+      currentUser,
+      email,
+    );
+  }
+
+  @Mutation(() => Boolean)
+  async deleteWorkspaceTrustedDomain(
+    @Args('input') { id }: DeleteTrustedDomainInput,
+    @AuthWorkspace() currentWorkspace: Workspace,
+  ): Promise<boolean> {
+    await this.workspaceTrustedDomainService.deleteTrustedDomain(
```

Comment:
When we delete a workspace do we remove the SSL certificate in Cloudflare? Since they are charged at a monthly cost and we have only 5000 slots available on the non-Enterprise plan we want to make sure we cleanup unused domains certificates when workspace are deleted by the subscription expiry cron or when someone manually removes an approved domain. Can be done as a separate issue

Reply:
Sorry I'm confusing with the custom domain feature, it's too early 😅. Did we take care of this for the custom domain feature?

Reply:
You're right I will do it in another PR